ci/format_pre.sh

```diff
@@ -39,9 +39,22 @@ trap_errors () {
     set -v
 }
 
+test_versions () {
+    local bazel_version repo_version
+    bazel_version="$(grep -E '^VERSION =' versions.bzl | cut -d= -f2 | xargs)"
+    repo_version="$(cat VERSION)"
+    if [[ "$bazel_version" != "$repo_version" ]]; then
+        echo "##[error]: Versions do not match, versions.bzl=${bazel_version} VERSION=${repo_version}" >&2
```

Comment:
Can we just make `versions.bzl` canonical and use `genrule` to generate `VERSION`?

Reply:
possibly we can make `versions.bzl` canonical - not sure what the benefit of generating `VERSION` would then be - most likely we can just pass the version in any rules rather than passing the file

i think the big downside to doing this is that `VERSION` is explicit and ~expected - but the obv upside is not having to keep 2 sources of authority (in sync)

Reply:
Maybe we can symlink `versions.bzl` to `VERSION`? I know this breaks down on some platforms with git though.

Reply:
im thinking to move this forward to make the `VERSION` file into a genrule, as ~suggested

ill go through the code and look for any non-bazel uses of it as i think i only recently added its use in bazel, so im pretty sure its used in other ways

not sure if there is anything externally that is reading that file

Reply:
thinking about this further @htuch - and doing 

```console
$ git grep VERSION | grep -v "_VERSION"
```

im thinking removing the existing `VERSION` file is a bad idea its linked to, referred to and used in scripts

im also concerned that others might use it for getting/setting build versions in other ways too

Reply:
...some of the `pkg_rules` use the `name` field such that we need it in order to have packages with the version in the name